src/screens/ProfileScreen/elements/TextWithEmoji.tsx

```diff
@@ -4,12 +4,22 @@ import emojiRegex from 'emoji-regex';
 import {COLORS} from '../../../utils/theme';
 import {fonts} from '../../../utils/fonts';
 
-export function TextWithEmoji({text}: {text: string}) {
+export function TextWithEmoji({
+  text,
+  textStyle,
+  testId
+}: {
+  text: string;
+  textStyle: object;
+  testId: string;
+}) {
   return (
-    <Text style={styles.text}>
-      {text.split(' ').map((item: string, index: number) => {
+    <Text style={[styles.text, textStyle]} testID={testId}>
+      {text?.split(' ').map((item: string, index: number) => {
         return (
-          <Text key={index} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
+          <Text
+            key={index}
+            style={[item.match(emojiRegex()) ? styles.emoji : styles.text, textStyle]}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here splits the text by spaces and then maps each word to a new Text component. This could lead to performance issues if the text is very long, as it creates a new Text component for each word. A better approach might be to use a library that can parse the text and emojis without needing to split and map the text manually.

Also, using the index as a key in a list is generally not recommended unless the list is static and does not change. If items are added or removed from the list, this can cause issues with React's rendering. If the text is unique, it could be used as the key instead.
```diff
- <Text key={index} style={[item.match(emojiRegex()) ? styles.emoji : styles.text, textStyle]}>
+ <Text key={item} style={[item.match(emojiRegex()) ? styles.emoji : styles.text, textStyle]}>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->